api/document/js/__tests__/friendly-error.test.ts

```diff
@@ -0,0 +1,98 @@
+import { makeErrorFriendly } from '../friendly-error';
+
+describe('makeErrorFriendly()', () => {
+  it('converts simple "detail" errors', () => {
+    const detail = ('XML syntax error - expected \'>\', line 4, ' +
+                    'column 3 (<string>, line 4)');
+    expect(makeErrorFriendly({ detail })).toBe(detail);
+  });
+
+  it('returns string errors as-is', () => {
+    const msg = 'NameError at /document/M-16-19\nname \'boom\' is not ...';
+    expect(makeErrorFriendly(msg)).toBe(msg);
+  });
+
+  it('returns other errors as their JSON stringification', () => {
+    expect(makeErrorFriendly(null)).toBe('null');
+  });
+
+  it('provides details on XML errors w/ invalid content types', () => {
+    /* tslint:disable */
+    const err = {
+      "children": [
+        {
+          "content": [
+            {
+              "content_type": "'k' is an invalid content type.",
+              "_sourceline": "3"
+            },
+            {}
+          ],
+          "_sourceline": "2"
+        }
+      ],
+      "_sourceline": "1"
+    };
+    /* tslint:enable */
+    expect(makeErrorFriendly(err)).toBe(
+      `In an element starting at line 3:\n` +
+      `* content_type - 'k' is an invalid content type.`,
```

Comment:
This can be improved in the future, since `content_type` is really the tag name when it comes to XML.